packages/buefy-next/src/components/rate/Rate.vue

```diff
@@ -37,14 +37,16 @@
     </div>
 </template>
 
-<script>
+<script lang="ts">
+import { defineComponent } from 'vue'
+import type { ExtractComponentProps } from '../../utils/helpers'
 import config from '../../utils/config'
 import Icon from '../icon/Icon.vue'
 
-export default {
+const Rate = defineComponent({
     name: 'BRate',
     components: {
-        [Icon.name]: Icon
+        [Icon.name as string]: Icon
```

Comment:
To perform type-checking for `Icon`, we have to "directly" name the component like the following:
```ts
components: {
    BIcon
}
```

Reply:
Now that I think about it this makes sense.